src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -510,14 +562,180 @@ internal void ReadToken(string encodedJson)
                 }
             }
 
-            EncodedToken = encodedJson;
+            _encodedToken = encodedJson;
+        }
+
+#if NET6_0_OR_GREATER
+        /// <summary>
+        /// Converts a span into an instance of <see cref="JsonWebToken"/>.
+        /// </summary>
+        /// <param name="encodedTokenMemory">A span representing a 'JSON Web Token' (JWT) in JWS or JWE Compact Serialization Format.</param>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedTokenMemory"/> is malformed, a valid JWT should have either 2 dots (JWS) or 4 dots (JWE).</exception>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedTokenMemory"/> does not have an non-empty authentication tag after the 4th dot for a JWE.</exception>
+        /// <exception cref="SecurityTokenMalformedException">if <paramref name="encodedTokenMemory"/> has more than 4 dots.</exception>
+        internal void ReadToken(ReadOnlyMemory<char> encodedTokenMemory)
+        {
+            // JWT must have 2 dots for JWS or 4 dots for JWE (a.b.c.d.e)
+            ReadOnlySpan<char> encodedTokenSpan = encodedTokenMemory.Span;
+
+            Dot1 = encodedTokenSpan.IndexOf("."); 
```

Comment:
I believe it is more performant to check for '.', than ".".
char vs. string.

Reply:
Replaced "." with '.'